src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -286,23 +299,33 @@ def _rbd_image(cls, ioctx, pool_name, namespace, image_name):
             # snapshots
             stat['snapshots'] = []
             for snap in img.list_snaps():
+                if img.mirror_image_get_mode() == rbd.RBD_MIRROR_IMAGE_MODE_JOURNAL:
```

Comment:
What happens if a user modifies the mirror mode during the execution of this code: it's checked multiple times fro the journal mode, but not for the snapshot... So even if checking/not checking it multiple times could lead to an error, we should stick to one approach and ensure the behaviour, even erronous, is the same for both branches (journal or snapshot). Otherwise, bugfixing this would be very tricky, since race conditions here would behave differently depending on the mode.

Reply:
I've extracted this to a variable so we only call it once. If we get a race condition with only one call then we can't do much I'd say.